Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -210,21 +212,66 @@ void* PcapLiveDevice::statsThreadMain(void *ptr)
 	return 0;
 }
 
+pcap_t* PcapLiveDevice::doOpen(DeviceMode mode)
+{
+	char errbuf[PCAP_ERRBUF_SIZE] = {'\0'};
+	pcap_t* pcap = pcap_create(m_Name, errbuf);

```

Comment:
Why did you break `pcap_open_live` into all of these calls to libpcap?
Why not keeping `pcap_open_live` and then call `pcap_set_immediate_mode`?

Reply:
The reason is that `pcap_open_live` creates already activated pcap handle, but `pcap_set_immediate_mode` should be called for non-active instance. Otherwise, it returns `PCAP_ERROR_ACTIVATED ` error.
Please, refer to http://www.tcpdump.org/manpages/pcap_set_immediate_mode.3pcap.html

Reply:
Oh ok I didn't know that. Thanks!